File: scoring.py
class Score:
    def __init__(self):
        self.combo = 0
        self.health = 100
        self.points = 0

    def hit(self):
        self.combo += 1
        self.points += 300
        self.health = min(100, self.health + 2)

    def miss(self):
        self.combo = 0
        self.health -= 10

File: osu_web/engine/scoring.js
export class Score {
  constructor(){ this.combo=0; this.points=0; this.health=100; }
  hit(value=300){ this.combo++; this.points+=value; this.health=Math.min(100,this.health+2); }
  miss(penalty=10){ this.combo=0; this.health-=penalty; }
  reset(){ this.combo=0; this.points=0; this.health=100; }
}
